js_modules/dagit/packages/core/src/assets/AssetsCatalogTable.tsx

```diff
@@ -10,48 +10,95 @@ import {
   useMergedRefresh,
   useQueryRefreshAtInterval,
 } from '../app/QueryRefresh';
+import {PythonErrorFragment} from '../app/types/PythonErrorFragment';
 import {toGraphId, tokenForAssetKey} from '../asset-graph/Utils';
 import {useLiveDataForAssetKeys} from '../asset-graph/useLiveDataForAssetKeys';
-import {useDocumentTitle} from '../hooks/useDocumentTitle';
 import {useQueryPersistedState} from '../hooks/useQueryPersistedState';
+import {AssetGroupSelector} from '../types/globalTypes';
 import {LoadingSpinner} from '../ui/Loading';
 import {StickyTableContainer} from '../ui/StickyTableContainer';
 
-import {AssetTable, ASSET_TABLE_FRAGMENT} from './AssetTable';
+import {AssetTable, ASSET_TABLE_DEFINITION_FRAGMENT, ASSET_TABLE_FRAGMENT} from './AssetTable';
 import {AssetViewModeSwitch} from './AssetViewModeSwitch';
 import {AssetsEmptyState} from './AssetsEmptyState';
 import {AssetKey} from './types';
+import {
+  AssetCatalogGroupTableQuery,
+  AssetCatalogGroupTableQueryVariables,
+} from './types/AssetCatalogGroupTableQuery';
 import {
   AssetCatalogTableQuery,
   AssetCatalogTableQuery_assetsOrError_AssetConnection_nodes,
 } from './types/AssetCatalogTableQuery';
+import {AssetTableFragment} from './types/AssetTableFragment';
 import {useAssetView} from './useAssetView';
 
 const PAGE_SIZE = 50;
 
 type Asset = AssetCatalogTableQuery_assetsOrError_AssetConnection_nodes;
 
-export const AssetsCatalogTable: React.FC<{prefixPath?: string[]}> = ({prefixPath = []}) => {
-  const [cursor, setCursor] = useQueryPersistedState<string | undefined>({queryKey: 'cursor'});
-  const [search, setSearch] = useQueryPersistedState<string | undefined>({queryKey: 'q'});
-  const [view, _setView] = useAssetView();
-
-  useDocumentTitle(
-    prefixPath && prefixPath.length ? `Assets: ${prefixPath.join(' \u203A ')}` : 'Assets',
-  );
-
+function useAllAssets(
+  groupSelector?: AssetGroupSelector,
+): {
+  query: QueryResult;
+  assets: AssetTableFragment[] | undefined;
+  error: PythonErrorFragment | undefined;
+} {
   const assetsQuery = useQuery<AssetCatalogTableQuery>(ASSET_CATALOG_TABLE_QUERY, {
+    skip: !!groupSelector,
     notifyOnNetworkStatusChange: true,
   });
+  const groupQuery = useQuery<AssetCatalogGroupTableQuery, AssetCatalogGroupTableQueryVariables>(
+    ASSET_CATALOG_GROUP_TABLE_QUERY,
+    {
+      skip: !groupSelector,
+      variables: {group: groupSelector},
+      notifyOnNetworkStatusChange: true,
+    },
+  );
+
+  return React.useMemo(() => {
+    if (groupSelector) {
+      return {
+        query: groupQuery,
+        error: undefined,
+        assets: groupQuery.data?.assetNodes.map<AssetTableFragment>((definition) => ({
+          __typename: 'Asset',
+          id: definition.id,
+          key: definition.assetKey,
+          definition,
+        })),
```

Comment:
Seems a little strange to reconstruct fragment objects from a GraphQL response. Why does this need to be remapped?

Reply:
Ahh this is a bit weird. I pulled this mapping out into a helper method and put a comment on it for now - I think that the core problem is that one query gets us `assets` (with an optional definition) and the other gets us `assetNodes` (that are a definition). 

I think the best path might be to make a separate interface for the AssetCatalogTable and convert both into that type, or change the backend so that we can fetch `assets(groupName: )` instead of `assetNodes(groupName:)`. Either way will revisit in a bit because I think it could be a bit messy!